rebasehelper/specfile.py

```diff
@@ -391,13 +392,19 @@ def _process_patches(self, comment_out=None, remove_patches=None, disable_inappl
                         prep.insert(i, '# The following patch contains conflicts')
                         comment_out.remove(num)
                         i += 1
+                        removed += 1
                         break
                 for num in reversed(remove_patches):
                     if num == index:
                         del prep[i]
                         remove_patches.remove(num)
                         i -= 1
+                        removed += 1
                         break
+                # When combining Patch tags and %patchlist, if a Patch is removed, the indexes
+                # of %patchlist patches change and %patch macros need to be modified.
+                if self.tag('Patch{}'.format(index)).section_name.startswith('%patchlist'):
+                    prep[i] = patch_re.sub(r'%patch{}\2'.format(index - removed), prep[i])
```

Comment:
Are you sure this works as expected? I'd like to see a test that proves it.

Reply:
I've tried it on a simple case (some Patch tags and a patchlist and a patch defined by Patch tag was removed) and it worked correctly. I couldn't think of a more complicated scenario.

As for testing, I'd really like to write some tests, the problem is that we are running tests even for the older version of rpm. And since we need access to tags, the test spec has to be parseable by rpm... I may try to solve it by mocking.. Or perhaps we could create a separate spec file just for this test and run it only on rpm version >= 4.15?

Reply:
> some Patch tags and a patchlist and a patch defined by Patch tag was removed

And what if a tag defined by %patchlist is removed?

> Or perhaps we could create a separate spec file just for this test and run it only on rpm version >= 4.15?

No, mocking is the correct way here.